lighthouse-core/audits/byte-efficiency/render-blocking-resources.js

```diff
@@ -60,6 +60,31 @@ function getNodesAndTimingByUrl(nodeTimings) {
   return urlMap;
 }
 
+/**
+ * Any stack specific timing overrides should go in this function.
+ * https://github.com/GoogleChrome/lighthouse/issues/2832#issuecomment-591066081
+ *
+ * @param {Node} node
+ * @param {LH.Gatherer.Simulation.NodeTiming} nodeTiming
+ * @param {LH.Artifacts.DetectedStack[]} Stacks
+ */
+function computeStackSpecificTiming(node, nodeTiming, Stacks) {
+  const stackSpecificTiming = {...nodeTiming};
+  if (!Stacks) return stackSpecificTiming;
+  if (Stacks.some(stack => stack.id === 'amp')) {
+    // AMP will load a linked stylesheet asynchronously if it has not been loaded after 2.1 seconds:
+    // https://github.com/ampproject/amphtml/blob/8e03ac2f315774070651584a7e046ff24212c9b1/src/font-stylesheet-timeout.js#L54-L59
+    // Any potential savings for AMP stylesheet nodes must therefore be capped at 2.1 seconds.
```

Comment:
perhaps I'm reading their code wrong, but doesn't the referenced AMP file cap at 2.1 seconds *since start of navigation* rather than a 2.1s *duration* for each individual stylesheet?

That first sentence in your comment is right based on my understanding, but I'm not sure this "therefore" is.

apologies if you got this idea from me, I didn't fully explain what their code does in my last comment 😅

Reply:
Would just removing the second comment make sense to you?

Reply:
Well the second comment describes the implementation. If the implementation is fixed along with it then sure :)

Reply:
Wait so we are capping the end time of each node to 2.1s rather than capping the duration?  The current logic caps the duration.

Reply:
Answer: yes.  Took me a while to fully comprehend.